generic/MAPL_Generic.F90

```diff
@@ -8544,19 +8557,30 @@ subroutine MAPL_GetResource_array(state, vals, label, default, rc)
                vals = default
             end select
          else
-            call ESMF_ConfigGetAttribute(state%cf, valuelist = vals, count = count, label = label_to_use, rc = status)
-            _VERIFY(status)
+            call ESMF_ConfigGetAttribute(config, valuelist = vals, count = count, label = label, _RC)
          end if
-         class default
+      class default
          _FAIL( "Unsupported type")
       end select
 
-      _RETURN(ESMF_SUCCESS)
+      ! Need a printer for arrays
+      call ESMF_ConfigGetAttribute(config, printrc, label = 'PRINTRC:', default = 0, _RC)
+
+      ! Can set printrc to negative to not print at all
+      if (MAPL_AM_I_Root() .and. printrc >= 0) then
+         if (label_is_present) then
+            label_to_print = label
+         else
+            label_to_print = trim(label)
+         end if
+         call print_resource_array(printrc, label_to_print, vals, default=default,_RC)
+      end if
 
-   end subroutine MAPL_GetResource_array
+      _RETURN(ESMF_SUCCESS)
 
+   end subroutine MAPL_GetResourceFromConfig_array
 
-   subroutine print_resource(printrc, label, val, default, rc)
+   subroutine print_resource_scalar(printrc, label, val, default, rc)
       integer, intent(in) :: printrc
```

Comment:
printrc does not appear to be used in this procedure?   I was looking because I wanted to see if there is something better than using an integer as an on/off switch.